homeassistant/components/mqtt/siren.py

```diff
@@ -0,0 +1,357 @@
+                if json_payload[STATE] == self._state_off:
+                    self._attr_is_on = False
+                del json_payload[STATE]
+
+            if json_payload:
+                # process attributes
+                try:
+                    vol.All(TURN_ON_SCHEMA)(json_payload)
+                except vol.MultipleInvalid as invalid_siren_parameters:
+                    _LOGGER.warning(
+                        "Unable to update siren state attributes from payload '%s': %s",
+                        json_payload,
+                        invalid_siren_parameters,
+                    )
+                    return
+            self._update(process_turn_on_params(self, json_payload))
+            self.async_write_ha_state()
+
+        if self._config.get(CONF_STATE_TOPIC) is None:
+            # Force into optimistic mode.
+            self._optimistic = True
+        else:
+            self._sub_state = await subscription.async_subscribe_topics(
+                self.hass,
+                self._sub_state,
+                {
+                    CONF_STATE_TOPIC: {
+                        "topic": self._config.get(CONF_STATE_TOPIC),
+                        "msg_callback": state_message_received,
+                        "qos": self._config[CONF_QOS],
+                        "encoding": self._config[CONF_ENCODING] or None,
+                    }
+                },
+            )
+
+    @property
+    def assumed_state(self):
+        """Return true if we do optimistic updates."""
+        return self._optimistic
+
+    @property
+    def extra_state_attributes(self) -> dict:
+        """Return the state attributes."""
+        mqtt_attributes = super().extra_state_attributes
+        attributes = (
+            copy.deepcopy(mqtt_attributes) if mqtt_attributes is not None else {}
+        )
+        attributes.update(self._attr_extra_state_attributes)
+        return attributes
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        return self._supported_features
+
+    async def async_conditional_publish(
+        self,
+        topic: str,
+        template: str,
+        value: Any,
+        variables: dict[str, Any],
+    ) -> None:
+        """Publish MQTT payload with command template if a topic is configured."""
+        payload = self._command_templates[template](value, variables)
+        if payload and payload not in PAYLOAD_NONE:
+            await mqtt.async_publish(
+                self.hass,
+                self._config[topic],
+                payload,
+                self._config[CONF_QOS],
+                self._config[CONF_RETAIN],
+                self._config[CONF_ENCODING],
+            )
+
+    async def async_turn_on(self, **kwargs) -> None:
+        """Turn the siren on.
+
+        This method is a coroutine.
+        """
+        await self.async_conditional_publish(
+            CONF_COMMAND_TOPIC,
+            CONF_COMMAND_TEMPLATE,
```

Comment:
Is it maybe a bit illogical that we can parse a JSON message without specifying a template, but we can't send one?

Reply:
We could produce JSON by default and remove the `payload_on` or `payload_off` config parameters. `command_template` and `command_template_off` could replace `payload_on` and `payload_off` and can be used when the standard JSON output needs to be customized.

Reply:
If have updated the code so that publishing JSON is the default. `payload_on` and `payload_off` are still used to assign the state to publish.